sdk/selfhelp/arm-selfhelp/samples/v2-beta/typescript/src/discoverySolutionListSample.ts

```diff
@@ -10,55 +10,33 @@
  */
 async function listDiscoverySolutionsAtResourceScope() {
-  const scope =
-    "subscriptions/0d0fcd2e-c4fd-4349-8497-200edb3923c6/resourcegroups/myresourceGroup/providers/Microsoft.KeyVault/vaults/test-keyvault-non-read";
   const filter = "ProblemClassificationId eq 'SampleProblemClassificationId1'";
   const options: DiscoverySolutionListOptionalParams = { filter };
   const credential = new DefaultAzureCredential();
   const client = new HelpRP(credential);
   const resArray = new Array();
-  for await (let item of client.discoverySolution.list(scope, options)) {
-    resArray.push(item);
-  }
-  console.log(resArray);
-}
-
-/**
- * This sample demonstrates how to Lists the relevant Azure diagnostics and solutions using [problemClassification API](https://learn.microsoft.com/rest/api/support/problem-classifications/list?tabs=HTTP)) AND  resourceUri or resourceType.<br/> Discovery Solutions is the initial entry point within Help API, which identifies relevant Azure diagnostics and solutions. <br/><br/> Required Input :  problemClassificationId (Use the [problemClassification API](https://learn.microsoft.com/rest/api/support/problem-classifications/list?tabs=HTTP)) <br/>Optional input: resourceUri OR resource Type <br/><br/> <b>Note: </b>  ‘requiredInputs’ from Discovery solutions response must be passed via ‘additionalParameters’ as an input to Diagnostics and Solutions API.
- *
- * @summary Lists the relevant Azure diagnostics and solutions using [problemClassification API](https://learn.microsoft.com/rest/api/support/problem-classifications/list?tabs=HTTP)) AND  resourceUri or resourceType.<br/> Discovery Solutions is the initial entry point within Help API, which identifies relevant Azure diagnostics and solutions. <br/><br/> Required Input :  problemClassificationId (Use the [problemClassification API](https://learn.microsoft.com/rest/api/support/problem-classifications/list?tabs=HTTP)) <br/>Optional input: resourceUri OR resource Type <br/><br/> <b>Note: </b>  ‘requiredInputs’ from Discovery solutions response must be passed via ‘additionalParameters’ as an input to Diagnostics and Solutions API.
- * x-ms-original-file: specification/help/resource-manager/Microsoft.Help/preview/2023-09-01-preview/examples/ListDiscoverySolutionsAtSubscriptionScope.json
- */
-async function listDiscoverySolutionsAtSubscriptionScope() {
-  const scope = "subscriptions/0d0fcd2e-c4fd-4349-8497-200edb3923c6";
-  const filter = "ProblemClassificationId eq 'SampleProblemClassificationId1'";
-  const options: DiscoverySolutionListOptionalParams = { filter };
-  const credential = new DefaultAzureCredential();
-  const client = new HelpRP(credential);
-  const resArray = new Array();
-  for await (let item of client.discoverySolution.list(scope, options)) {
+  for await (let item of client.discoverySolution.list(options)) {
     resArray.push(item);
   }
   console.log(resArray);
 }
 
 async function main() {
   listDiscoverySolutionsAtResourceScope();
```

Comment:
@Alancere - Does the naming of the method matter here? Because the DiscoverySolution API is now a tenant level API. I see in the code it says "listDiscoverySolutionsAtResourceScope". Although, the implementation of it is correct.